blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/datastax/DBasicMetricsRW.java

```diff
@@ -0,0 +1,350 @@
+
+    private boolean areStringMetricsDropped = Configuration.getInstance().getBooleanProperty( CoreConfig.STRING_METRICS_DROPPED);
+    private List<String> tenantIdsKept = Configuration.getInstance().getListProperty(CoreConfig.TENANTIDS_TO_KEEP);
+    private Set<String> keptTenantIdsSet = new HashSet<String>(tenantIdsKept);
+
+    private DRawIO rawIO = new DRawIO();
+    private LocatorIO locatorIO = IOContainer.fromConfig().getLocatorIO();
+
+    /**
+     * This method inserts a collection of {@link com.rackspacecloud.blueflood.types.IMetric} objects
+     * to the appropriate Cassandra column family.
+     *
+     * @param metrics
+     *
+     * @throws IOException
+     */
+    @Override
+    public void insertMetrics( Collection<IMetric> metrics ) throws IOException {
+        // if there are strings & booleans in this request, they will be tracked as numeric metrics.
+        // not sure how we want to get around that.
+        Timer.Context ctx = Instrumentation.getWriteTimerContext( CassandraModel.CF_METRICS_FULL_NAME );
+
+        Map<Locator, ResultSetFuture> futures = new HashMap<Locator, ResultSetFuture>();
+
+        try {
+
+            for( IMetric metric : metrics ) {
+
+                if (!shouldPersist(metric)) {
+                    LOG.trace("Metric shouldn't be persisted, skipping insert", metric.getLocator().toString());
+                    continue;
+                }
+
+                Locator locator = metric.getLocator();
+
+                if( !isLocatorCurrent( locator ) ) {
+
+                    setLocatorCurrent( locator );
+
+                    if( !DataType.isStringOrBoolean( metric.getMetricValue() ) )
+                        locatorIO.insertLocator( locator );
+                }
+
+
+                futures.put( locator, rawIO.insertAsync( metric ) );
+            }
+
+            for( Map.Entry<Locator, ResultSetFuture> f : futures.entrySet() ) {
+
+                try {
+                    ResultSet result = f.getValue().getUninterruptibly();
+                    LOG.trace( "result.size=" + result.all().size() );
+                }
+                catch ( Exception e ) {
+                    Instrumentation.markWriteError();
+                    LOG.error(String.format("error writing metric for locator %s",
+                            f.getKey()), e );
+                }
+            }
+        }
+        finally {
+
+            ctx.close();
+        }
+    }
+
+    /**
+     * Fetches {@link com.rackspacecloud.blueflood.outputs.formats.MetricData} objects for the
+     * specified {@link com.rackspacecloud.blueflood.types.Locator} and
+     * {@link com.rackspacecloud.blueflood.types.Range} from the specified column family
+     *
+     * @param locator
+     * @param range
+     * @param gran
+     * @return
+     */
+    @Override
+    public MetricData getDatapointsForRange( final Locator locator, Range range, Granularity gran ) throws IOException {
```

Comment:
Isn't this a dup of DAbstractMetricsRW.getDatapointsForRange() ? I couldn't tell the diff.